src/test/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStepTest.java

```diff
@@ -136,4 +140,28 @@
         assertNull(b3.getPreviousBuild());
     }
 
+    @Issue("JENKINS-34547")
+    @Test public void concurrentBuildProperty() throws Exception {
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition("properties([[$class: 'DisableConcurrentBuildsJobProperty']])\n"
+                + "semaphore 'hang'"));
+
+        assertTrue(p.isConcurrentBuild());
+
+        WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+        SemaphoreStep.waitForStart("hang/1", b1);
+        assertFalse(p.isConcurrentBuild());
+
+        QueueTaskFuture<WorkflowRun> futureB2 = p.scheduleBuild2(0);
+        assertFalse(futureB2.getStartCondition().isDone());
```

Comment:
I suspect this would pass regardless, as queue maintenance may be asynchronous and you have not waited to see if the build might start a bit later.